middleware/http/oauth2/oauth2_middleware.go

```diff
@@ -81,6 +82,9 @@ func (m *Middleware) GetHandler(metadata middleware.Metadata) (func(h fasthttp.R
 			} else {
 				authState := session.GetString(savedState)
 				redirectURL := session.GetString(redirectPath)
+				if meta.ForceHTTPS == "true" {
+					redirectURL = "https://" + string(ctx.Request.Host()) + redirectURL
+				}
```

Comment:
Please add unit-test for this change.

Reply:
Working on it.

Reply:
Hi @youngbupark. Could you please give me advice on how unit test for this should be done? This is a Authorization Code Flow of OAuth2 protocol which should be done via browser to pass user credentials directly to Authorization Provider without getting them to the app. When I was testing it locally I've setup whole infrastructure in order to do that. There is no existing unit tests for this middleware, so I can't mimic existing tests or extend them. Maybe I'm looking at the wrong direction and don't see some obvious ways to do this. Advice or help would be appreciated.

Reply:
Thanks for your investigation and your time. The initial ask is to test the below code by separating it as separate method or using mock:

```
redirectURL := session.GetString(redirectPath)
				if meta.ForceHTTPS == "true" {
					redirectURL = "https://" + string(ctx.Request.Host()) + redirectURL
				}
```

If it is difficult to write the test without the actual backend service support and you validate it locally, I am ok to merge it now.  @yaron2 any objection?

JFYI - We are planning to have conformance test. we will provide the common test infra for middleware test later. @mukundansundar